src/bcf/mod.rs

```diff
@@ -1296,4 +1298,32 @@ mod tests {
 
         assert_eq!(rec.info(b"X").string().unwrap().unwrap().len(), 2);
     }
+
+    #[test]
+    fn test_alt_allele_dosage() {
+        let path = &"test/test_string.vcf";
+        let mut bcf = Reader::from_path(path).expect("Error opening file.");
+        let header = bcf.header();
+
+        for (_i, record_result) in bcf.records().enumerate() {
+            let mut record = record_result.expect("Fail to read record");
+            let mut s = String::new();
+
+            let sample_count = usize::try_from(record.sample_count()).unwrap();
+            let mut n_ref = vec![0; sample_count];
+            let mut n_alt = vec![0; sample_count];
+            let gts = record.genotypes().expect("Error reading genotypes");
+            for sample_index in 0..sample_count {
+                // for each sample
+                for gta in gts.get(sample_index).iter() {
+                    // for each allele
+                    match gta.index() {
+                        Some(0) => n_ref[sample_index] += 1, // reference allele
+                        Some(_) => n_alt[sample_index] += 1, // alt allele
+                        None => break,                       // missing allele
+                    }
+                }
+            }
+        }
+    }
```

Comment:
Great idea adding this as a test case, as well. But this should then probably include some [`assert_eq!()`](https://doc.rust-lang.org/std/macro.assert_eq.html) invocations at the end to check that the right counts were generated?